<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="author" content="Erick Matsen">
  <title>Why phylogenetics?</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="reveal.js/css/reveal.css">
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <link rel="stylesheet" href="reveal.js/css/theme/simple.css" id="theme">
  <!-- Printing and PDF exports -->
  <script>
    var link = document.createElement( 'link' );
    link.rel = 'stylesheet';
    link.type = 'text/css';
    link.href = window.location.search.match( /print-pdf/gi ) ? 'reveal.js/css/print/pdf.css' : 'reveal.js/css/print/paper.css';
    document.getElementsByTagName( 'head' )[0].appendChild( link );
  </script>
  <!--[if lt IE 9]>
  <script src="reveal.js/lib/js/html5shiv.js"></script>
  <![endif]-->
  <style type="text/css">
      .reveal { font-size: 32px; }
      .reveal h1 { font-size: 1.35em; }
  
      .reveal section img {
          border: none;
          box-shadow: none;
          vertical-align: middle;
      }
  
      .reveal h1,
      .reveal h2,
      .reveal h3,
      .reveal h4,
      .reveal h5,
      .reveal h6 {
        font-family: "News Cycle", Lato, sans-serif;
        text-shadow: none; }
  
      div.bigpurple {
        font-size: 3em;
        line-height: 1em;
        color: purple;
      }
  
      div.caption {
        font-size: 24px;
        line-height: 1em;
      }
  
      div.citation {
          position: absolute;
          bottom:0px;
          right:0px;
          z-index:100;
          font-size: 0.45em !important;
      }
  
  </style>
</head>
<body>
  <div class="reveal">
    <div class="slides">

<section id="title-slide">
  <h1 class="title">Why phylogenetics?</h1>
  <p class="author">Erick Matsen</p>
</section>

<section id="popular-image-of-phylogenetics" class="slide level1">
<h1>Popular image of phylogenetics</h1>
<img src="figures/motivation/sponges-sister-tree.png" class="stretch">
<div class="caption">
Simion <em>et al</em>, <em>Current Biology</em>, 2017
</div>
</section>
<section id="what-can-we-do-with-phylogenetics" class="slide level1">
<h1>What can we <em>do</em> with phylogenetics?</h1>
<p>Examples today:</p>
<ol type="1">
<li>Felsenstein independent contrasts</li>
<li>Understanding flu transmission</li>
</ol>
</section>
<section id="section" class="slide level1">
<h1></h1>
<div class="bigpurple">
Independent contrasts
</div>
</section>
<section id="section-1" class="slide level1">
<h1></h1>
<blockquote class="twitter-tweet">
<p lang="en" dir="ltr">
Genome of the whale shark, Earth's largest fish—contributed by <a href="https://twitter.com/geochurch?ref_src=twsrc%5Etfw"><span class="citation" data-cites="geochurch">@geochurch</span></a> and colleagues. Ostensibly extremely significant correlations are reported between physiological and genomic features across taxa, interpreted as evincing scaling laws. <a href="https://t.co/9EvT18ClPR">https://t.co/9EvT18ClPR</a> (1/n)
</p>
— William DeWitt (<span class="citation" data-cites="wsdewitt">@wsdewitt</span>) <a href="https://twitter.com/wsdewitt/status/1291147602507862016?ref_src=twsrc%5Etfw">August 5, 2020</a>
</blockquote>
<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
</section>
<section id="section-2" class="slide level1">
<h1></h1>
<p><a href="https://www.pnas.org/content/pnas/117/34/20662/F2.large.jpg" target="_blank"><img src="https://www.pnas.org/content/pnas/117/34/20662/F2.large.jpg" height="620"></a></p>
<p><a href="https://www.pnas.org/content/117/34/20662">Paper link</a>. gray: Hyperoartia, Ascidiacea, Chromadorea, Insecta, and Saccharomycetes; turquoise: Chondrichthyes [cyan indicates whale shark]; light blue: Actinopterygii; aquamarine: Sarcopterygii; dark green: Amphibia; light green: Reptilia; dark yellow: Aves; orange: Mammalia</p>
</section>
<section id="section-3" class="slide level1">
<h1></h1>
<img width="970" src="figures/comparative-method/whale-shark-panel-d-split_nofonts.svg">
</section>
<section id="say-we-measure-two-traits" class="slide level1">
<h1>Say we measure two traits</h1>
<img src="figures/comparative-method/traits.png" class="stretch">
</section>
<section id="do-traits-x-and-y-look-correlated" class="slide level1">
<h1>Do traits <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span> look correlated?</h1>
<img src="figures/comparative-method/false-correlation-one-symbol.png" height="500">
</section>
<section id="do-traits-x-and-y-look-correlated-1" class="slide level1">
<h1>Do traits <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span> look correlated?</h1>
<img src="figures/comparative-method/false-correlation-two-symbols.png" height="500">
</section>
<section id="stars-from-left-branch-squares-from-the-right" class="slide level1">
<h1>Stars from left branch, squares from the right</h1>
<img src="figures/comparative-method/worst-case-tree.png" style="float:left" height="510"> <img src="figures/comparative-method/false-correlation-two-symbols.png" style="float:right" height="510">
<p><br></p>
<div class="fragment">
Need to correct for evolutionary history to test evolution hypothesis.
</div>
</section>
<section id="felsenstein-1985" class="slide level1">
<h1>Felsenstein 1985</h1>
<p>Felsenstein, J. (1985). Phylogenies and the Comparative Method. The American Naturalist, 125(1), 1–15. <a href="https://paperpile.com/shared/Vp8yHJ">[PDF]</a></p>
<div class="fragment">
<p>Huey, R. B., Garland, T., Jr, &amp; Turelli, M. (2019). Revisiting a Key Innovation in Evolutionary Biology: Felsenstein’s “Phylogenies and the Comparative Method.” The American Naturalist, 193(6), 755–772. <a href="https://doi.org/10.1086/703055">[DOI]</a></p>
</div>
</section>
<section id="section-4" class="slide level1">
<h1></h1>
<img width="970" src="figures/comparative-method/dunn-pairwise-felsenstein-simple_nofonts.svg">
<p>This paper blows big holes in these papers:</p>
<ul>
<li>Kryuchkova-Mostacci, N., &amp; Robinson-Rechavi, M. (2016). Tissue-Specificity of Gene Expression Diverges Slowly between Orthologs, and Rapidly between Paralogs. PLoS Computational Biology, 12(12), e1005274. <a href="https://doi.org/10.1371/journal.pcbi.1005274">[DOI]</a></li>
<li>Levin, M., Anavy, L., Cole, A. G., Winter, E., Mostov, N., Khair, S., … Yanai, I. (2016). The mid-developmental transition and the evolution of animal body plans. Nature, 531, 637. <a href="https://doi.org/10.1038/nature16994">[DOI]</a></li>
</ul>
<p>… don’t make this mistake, please!</p>
</section>
<section id="for-a-paper-that-does-it-right" class="slide level1">
<h1>For a paper that does it right…</h1>
<p>Schraiber, J. G., Mostovoy, Y., Hsu, T. Y., &amp; Brem, R. B. (2013). Inferring evolutionary histories of pathway regulation from transcriptional profiling data. PLoS Computational Biology, 9(10), e1003255. <a href="https://doi.org/10.1371/journal.pcbi.1003255">[DOI]</a></p>
<p><a href="https://www.youtube.com/watch?v=3lxqv_iJeLY" class="uri">https://www.youtube.com/watch?v=3lxqv_iJeLY</a></p>
</section>
<section id="understanding-phylogenetic-trees" class="slide level1">
<h1>Understanding phylogenetic trees</h1>
</section>
<section id="section-5" class="slide level1">
<h1></h1>
<div class="bigpurple">
Flu transmission patterns
</div>
</section>
<section id="where-does-the-flu-come-from" class="slide level1">
<h1>Where does the flu come from?</h1>
<img src="figures/phylogeography/bedford-2010-migration-map.png" class="stretch">
<div class="fragment">
<p>Bedford, T., Cobey, S., Beerli, P., &amp; Pascual, M. (2010). Global Migration Dynamics Underlie Evolution and Persistence of Human Influenza A (H3N2). PLoS Pathogens, 6(5), e1000918. <a href="https://doi.org/10.1371/journal.ppat.1000918">[DOI]</a></p>
</div>
</section>
<section id="infer-trunk-location" class="slide level1">
<h1>Infer trunk location</h1>
<img src="figures/phylogeography/bedford-2010-migration-tree.png" class="stretch">
</section>
<section id="infer-trunk-location-1" class="slide level1">
<h1>Infer trunk location</h1>
<img src="figures/phylogeography/bedford-2010-migration-trunk.png" class="stretch">
</section>
<section id="where-is-flu-going" class="slide level1">
<h1>Where is flu going?</h1>
<p><br></p>
<p>I.e. which virus is going to take over the population?</p>
</section>
<section id="local-branching-index" class="slide level1">
<h1>Local branching index</h1>
<img src="figures/motivation/lbi-definition.png" class="stretch">
</section>
<section id="local-branching-index-in-nextflu.org" class="slide level1">
<h1>Local branching index in nextflu.org</h1>
<img src="figures/motivation/lbi-nextflu_nofonts.svg" class="stretch">
</section>
<section id="try-it-out" class="slide level1">
<h1>Try it out</h1>
<p><a href="http://nextflu.org/" class="uri">http://nextflu.org/</a></p>
<p>Neher, R. A., &amp; Bedford, T. (2015). nextflu: Real-time tracking of seasonal influenza virus evolution in humans. Bioinformatics . <a href="https://doi.org/10.1093/bioinformatics/btv381">[DOI]</a></p>
<p>Neher, R. A., Russell, C. A., &amp; Shraiman, B. I. (2014). Predicting evolution from the shape of genealogical trees. eLife, 3. <a href="https://doi.org/10.7554/eLife.03568">[DOI]</a></p>
<p><a href="http://nextstrain.org/" class="uri">http://nextstrain.org/</a></p>
<p>Hadfield, Megill, Bell, Huddleston, Potter, Callender, …, Bedford, Neher, (2018). Nextstrain: real-time tracking of pathogen evolution. Bioinformatics. <a href="https://doi.org/10.1093/bioinformatics/bty407">[DOI]</a></p>
</section>
    </div>
  </div>

  <script src="reveal.js/lib/js/head.min.js"></script>
  <script src="reveal.js/js/reveal.js"></script>

  <script>

      // Full list of configuration options available at:
      // https://github.com/hakimel/reveal.js#configuration
      Reveal.initialize({
        // Display controls in the bottom right corner
        controls: false,
        // Push each slide change to the browser history
        history: true,
        // Transition style
        transition: 'none', // none/fade/slide/convex/concave/zoom
        math: {
          mathjax: 'MathJax/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // Optional reveal.js plugins
        dependencies: [
          { src: 'reveal.js/lib/js/classList.js', condition: function() { return !document.body.classList; } },
          { src: 'reveal.js/plugin/zoom-js/zoom.js', async: true },
          { src: 'reveal.js/plugin/math/math.js', async: true },
          { src: 'reveal.js/plugin/notes/notes.js', async: true }
        ]
      });
    </script>
    </body>
</html>
