<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="author" content="Erick Matsen">
  <title>Phylogenetics methods</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="reveal.js/css/reveal.css">
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <link rel="stylesheet" href="reveal.js/css/theme/simple.css" id="theme">
  <!-- Printing and PDF exports -->
  <script>
    var link = document.createElement( 'link' );
    link.rel = 'stylesheet';
    link.type = 'text/css';
    link.href = window.location.search.match( /print-pdf/gi ) ? 'reveal.js/css/print/pdf.css' : 'reveal.js/css/print/paper.css';
    document.getElementsByTagName( 'head' )[0].appendChild( link );
  </script>
  <!--[if lt IE 9]>
  <script src="reveal.js/lib/js/html5shiv.js"></script>
  <![endif]-->
  <style type="text/css">
      .reveal { font-size: 32px; }
      .reveal h1 { font-size: 1.35em; }
  
      .reveal section img {
          border: none;
          box-shadow: none;
          vertical-align: middle;
      }
  
      .reveal h1,
      .reveal h2,
      .reveal h3,
      .reveal h4,
      .reveal h5,
      .reveal h6 {
        font-family: "News Cycle", Lato, sans-serif;
        text-shadow: none; }
  
      div.bigpurple {
        font-size: 3em;
        line-height: 1em;
        color: purple;
      }
  
      div.caption {
        font-size: 24px;
        line-height: 1em;
      }
  
      div.citation {
          position: absolute;
          bottom:0px;
          right:0px;
          z-index:100;
          font-size: 0.45em !important;
      }
  
  </style>
</head>
<body>
  <div class="reveal">
    <div class="slides">

<section id="title-slide">
  <h1 class="title">Phylogenetics methods</h1>
  <p class="author">Erick Matsen</p>
  <p class="date">Thank you to: Trevor Bedford (Fred Hutch)</p>
</section>

<section id="section" class="slide level1">
<h1></h1>
<img src="figures/phylo/what_is_phylo_nofonts.svg" class="stretch">
</section>
<section id="section-1" class="slide level1">
<h1></h1>
<img src="figures/phylo/what_is_phylo_taxa_nofonts.svg" class="stretch">
</section>
<section id="section-2" class="slide level1">
<h1></h1>
<img src="figures/phylo/what_is_phylo_unrooted_nofonts.svg" class="stretch">
</section>
<section id="data-sequence-alignment" class="slide level1">
<h1>“data”: sequence alignment</h1>
<img src="figures/alignment/bigProtAln.png" class="stretch">
</section>
<section id="a-sequence-alignment-is-not-a-fact" class="slide level1">
<h1>A sequence alignment is not a fact</h1>
<img src="figures/alignment/bigProtAln.png" class="stretch">
<p>… it is a complex inference.</p>
</section>
<section id="in-molecular-phylogenetics-homology-means" class="slide level1">
<h1>In molecular phylogenetics, <em>homology</em> means…</h1>
<img src="figures/alignment/bigProtAln-column_nofonts.svg" class="stretch">
</section>
<section id="in-molecular-phylogenetics-homology-means-1" class="slide level1">
<h1>In molecular phylogenetics, <em>homology</em> means…</h1>
<img src="figures/alignment/bigProtAln-column_nofonts.svg" class="stretch">
<p>Thus sequence alignment for phylogenetics is <br> a <em>super challenging inference</em>. More about this later.</p>
</section>
<section id="types-of-phylogenetic-inference-methods" class="slide level1">
<h1>Types of phylogenetic inference methods</h1>
<ul>
<li>Distance-based</li>
<li>Parsimony</li>
<li>Likelihood-based
<ul>
<li>Maximum likelihood</li>
<li>Bayesian</li>
</ul></li>
</ul>
</section>
<section id="distance-based-phylogenetics" class="slide level1">
<h1>Distance-based phylogenetics</h1>
<img src="figures/phylo/what_is_phylo_dist_nofonts.svg" class="stretch">
</section>
<section id="distance-based-phylogenetics-1" class="slide level1">
<h1>Distance-based phylogenetics</h1>
<img src="figures/phylo/what_is_phylo_dist_nofonts.svg" class="stretch">
<p>Note that the matrix doesn’t have to come from sequence data.</p>
</section>
<section id="parsimony-phylogenetics" class="slide level1">
<h1>Parsimony phylogenetics</h1>
<p><br></p>
<p><a href="https://en.wikipedia.org/wiki/Maximum_parsimony_(phylogenetics)">Parsimony</a> is based on <a href="https://en.wikipedia.org/wiki/Occam&#39;s_razor">Occam’s razor</a></p>
<p>Among competing hypotheses that predict equally well, the one with the fewest assumptions should be selected.</p>
<p><br></p>
<div class="caption">
The next few slides are from Trevor Bedford.
</div>
</section>
<section id="section-3" class="slide level1">
<h1></h1>
<p>Say we have three viruses <img src="figures/parsimony/parsimony_1.png" height="620"></p>
</section>
<section id="section-4" class="slide level1">
<h1></h1>
<p>Say we have three viruses <img src="figures/parsimony/parsimony_2.png" height="620"></p>
</section>
<section id="section-5" class="slide level1">
<h1></h1>
<p>We can explain these sequences with 3 mutations <img src="figures/parsimony/parsimony_3.png" height="620"></p>
</section>
<section id="section-6" class="slide level1">
<h1></h1>
<p>This topology requires 3 mutations at minimum <img src="figures/parsimony/parsimony_4.png" height="620"></p>
</section>
<section id="section-7" class="slide level1">
<h1></h1>
<p><em>Exercise: which topology is more parsimonious?</em> <img src="figures/parsimony/parsimony_exercise_1.png" height="620"></p>
</section>
<section id="section-8" class="slide level1">
<h1></h1>
<p><em>Exercise: which topology is more parsimonious?</em> <img src="figures/parsimony/parsimony_exercise_2.png" height="620"></p>
</section>
<section id="section-9" class="slide level1">
<h1></h1>
<p><em>Exercise: which topology is more parsimonious?</em> <img src="figures/parsimony/parsimony_exercise_3.png" height="620"></p>
</section>
<section id="section-10" class="slide level1">
<h1></h1>
<p><em>Exercise: which topology is more parsimonious?</em> <img src="figures/parsimony/parsimony_exercise_4.png" height="620"></p>
</section>
<section id="section-11" class="slide level1">
<h1></h1>
<p><em>Exercise: which topology is more parsimonious?</em> <img src="figures/parsimony/parsimony_exercise_5.png" height="620"></p>
</section>
<section id="section-12" class="slide level1">
<h1></h1>
<p><em>Exercise: which topology is more parsimonious?</em> <img src="figures/parsimony/parsimony_exercise_6.png" height="620"></p>
</section>
<section id="section-13" class="slide level1">
<h1></h1>
<p><em>Exercise: which topology is more parsimonious?</em> <img src="figures/parsimony/parsimony_exercise_7.png" height="620"></p>
</section>
<section id="section-14" class="slide level1">
<h1></h1>
<p><em>Exercise: which topology is more parsimonious?</em> <img src="figures/parsimony/parsimony_exercise_8.png" height="620"></p>
</section>
<section id="section-15" class="slide level1">
<h1></h1>
<p>Parsimony seems sensible.</p>
<p>Is it the most popular phylogenetics method?</p>
<div class="fragment">
<p><em>No. There are situations in which the correct tree has more mutations.</em></p>
</div>
</section>
<section id="section-16" class="slide level1">
<h1></h1>
<img width="970" src="figures/parsimony/felsenstein-zone-title.png"> <img width="320" src="figures/parsimony/felsenstein-zone-tree.png">
</section>
<section id="long-branch-attraction" class="slide level1">
<h1>Long branch attraction</h1>
<img src="figures/parsimony/lba-tree-1.png" height="500">
<div class="citation">
http://slideplayer.com/slide/9059488/
</div>
</section>
<section id="long-branch-attraction-1" class="slide level1">
<h1>Long branch attraction</h1>
<img src="figures/parsimony/lba-tree-2.png" height="500">
<div class="citation">
http://slideplayer.com/slide/9059488/
</div>
</section>
<section id="are-we-cooked" class="slide level1">
<h1>Are we cooked?</h1>
<p><br></p>
<div class="fragment">
Given enough data, likelihood methods will converge to the true tree.
</div>
</section>
<section id="likelihood-setup" class="slide level1">
<h1>Likelihood setup</h1>
<ul>
<li>
Come up with a statistical model of experiment in terms of some data and some parameters
</li>
<li class="fragment">
Write down a <em>likelihood function</em> that expresses the probability of generating the observed data given those parameters
</li>
<li class="fragment">
Now we can evaluate likelihood under various parameter values
</li>
<li class="fragment">
We will write this likelihood <span class="math inline">\(\Pr(D \mid \theta)\)</span>. <br> <em>Why is this appropriate notation?</em>
</li>
</ul>
</section>
<section id="example-picking-peaches" class="slide level1">
<h1>Example: picking peaches</h1>
<p>Say <span class="math inline">\(p\)</span> is the probability of getting a ripe peach, and each draw is independent.</p>
<ul>
<li class="fragment">
What is the probability of getting two ripe peaches in a row?
</li>
<li class="fragment">
What about if after harvesting two peaches, we have one ripe one?
</li>
<li class="fragment">
What about if after harvesting 20 peaches, we have 6 ripe ones?
</li>
</ul>
</section>
<section id="example-picking-peaches-1" class="slide level1">
<h1>Example: picking peaches</h1>
<p>Say that, after harvesting 20 peaches, we have 6 ripe ones.</p>
<p>Say <span class="math inline">\(p\)</span> is the probability of getting a ripe peach, and each draw is independent. Model using the <a href="http://en.wikipedia.org/wiki/Binomial_distribution">binomial distribution</a>.</p>
<p> </p>
<p>The <em>likelihood</em> of getting the observed result is <span class="math display">\[
{ {20} \choose 6} \, p^6 \, (1-p)^{20-6}.
\]</span> Recall: <span class="math inline">\({ {20} \choose 6}\)</span> is the number of ways of choosing 6 items out of 20.</p>
</section>
<section id="peach-picking-likelihood-surface" class="slide level1">
<h1>Peach picking likelihood surface</h1>
<img src="figures/likelihood/ripe-likelihood_nofonts.svg" height="450">
<p>The <em>maximum likelihood estimate</em> of the parameter(s) of interest is the parameter value(s) that maximize the likelihood.</p>
<!--
#
<s> The *maximum likelihood estimate* of the parameter(s) of interest is the parameter value(s) that maximize the likelihood. </s>

<img src="figures/likelihood/prob-models-have-dials_nofonts.svg" class="stretch">

Fiddle with the dials until it looks good.
-->
</section>
<section id="questions" class="slide level1">
<h1>Questions</h1>
<img src="figures/likelihood/ripe-likelihood_nofonts.svg" height="250">
<ol>
<li class="fragment">
What is the maximum likelihood (ML) estimate of <span class="math inline">\(p\)</span> given our experiment?
</li>
<li class="fragment">
Would the result of this ML estimate be different if we got 60 ripe peaches out of 200?
</li>
<li class="fragment">
Intuitively, would the shape of the likelihood curve be different with this larger dataset?
</li>
</ol>
</section>
<section id="likelihood-recap" class="slide level1">
<h1>Likelihood recap</h1>
<ul>
<li>Maximum likelihood is a way of inferring unknown parameters</li>
<li>To apply likelihood, we need a <em>model</em> of the system under investigation</li>
<li>In general, the “likelihood” is the likelihood of generating the data under the given parameters, written <span class="math inline">\(P(D | \theta),\)</span> where <span class="math inline">\(D\)</span> is the data and <span class="math inline">\(\theta\)</span> are the parameters.</li>
</ul>
<!--
# Setup for likelihood based phylogenetics
The phylogenetic likelihood of a tree is the likelihood of generating the observed data given that tree (under the sequence evolution model).

<img src="figures/phylo/what_is_phylo_ML.scour_nofonts.svg" class="stretch">

Note that the UW's own [Joe Felsenstein](http://evolution.genetics.washington.edu/phylip/felsenstein.html) was the [first](http://www.springerlink.com/content/g2202t346n826461/) to formalize this and develop efficient algorithms.


# [Sequence evolution models](http://en.wikipedia.org/wiki/Substitution_model) tell us the probability of seeing a certain mutation in some period of (evolutionary) time
<div class="incremental">
* Nucleotide models are fit "on the fly"
    * e.g. F81, HKY, GTR

* Protein models are typically pre-made
    * e.g. [JTT](http://www.scopus.com/record/display.url?eid=2-s2.0-0026691182&origin=inward&txGid=u2307zIZ5me4AtASSSuc_3u%3a2) (Jones, Taylor, and Thornton), and [WAG](http://mbe.oxfordjournals.org/content/18/5/691.short) (Whelan and Goldman) matrices

* Codon models are a great idea
    * Position matters!
    * e.g. [SRD06](http://mbe.oxfordjournals.org/content/23/1/7.full) model
</div>


# Model hierarchy, from [Posada and Crandall](http://bioinformatics.oxfordjournals.org/cgi/reprint/14/9/817)
<img src="figures/phylo/hLRT_Posada_and_Crandal_bioInfo.png" class="stretch">


# Calculating likelihood of a single column
<img src="figures/phylo/column_likelihood_nofonts.svg" class="stretch">


# Likelihood of an alignment
<img src="figures/phylo/aln_likelihood.scour_nofonts.svg" class="stretch">
-->
</section>
<section id="crazy-but-typical-model-assumptions" class="slide level1">
<h1>Crazy but typical model assumptions</h1>
<ul>
<li>differences between sequences only appear by point mutation</li>
<li>evolution happens on each column independently</li>
<li>sequences are evolving according to <em>reversible</em> models (this excludes selection and directional evolution of base composition)</li>
<li>the evolutionary process is identical on all branches of the tree</li>
</ul>
</section>
    </div>
  </div>

  <script src="reveal.js/lib/js/head.min.js"></script>
  <script src="reveal.js/js/reveal.js"></script>

  <script>

      // Full list of configuration options available at:
      // https://github.com/hakimel/reveal.js#configuration
      Reveal.initialize({
        // Display controls in the bottom right corner
        controls: false,
        // Push each slide change to the browser history
        history: true,
        // Transition style
        transition: 'none', // none/fade/slide/convex/concave/zoom
        math: {
          mathjax: 'MathJax/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // Optional reveal.js plugins
        dependencies: [
          { src: 'reveal.js/lib/js/classList.js', condition: function() { return !document.body.classList; } },
          { src: 'reveal.js/plugin/zoom-js/zoom.js', async: true },
          { src: 'reveal.js/plugin/math/math.js', async: true },
          { src: 'reveal.js/plugin/notes/notes.js', async: true }
        ]
      });
    </script>
    </body>
</html>
