<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="author" content="Erick Matsen">
  <title>Sequence alignment from the   phylogenetic perspective</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="reveal.js/css/reveal.css">
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <link rel="stylesheet" href="reveal.js/css/theme/simple.css" id="theme">
  <!-- Printing and PDF exports -->
  <script>
    var link = document.createElement( 'link' );
    link.rel = 'stylesheet';
    link.type = 'text/css';
    link.href = window.location.search.match( /print-pdf/gi ) ? 'reveal.js/css/print/pdf.css' : 'reveal.js/css/print/paper.css';
    document.getElementsByTagName( 'head' )[0].appendChild( link );
  </script>
  <!--[if lt IE 9]>
  <script src="reveal.js/lib/js/html5shiv.js"></script>
  <![endif]-->
  <style type="text/css">
      .reveal { font-size: 32px; }
      .reveal h1 { font-size: 1.35em; }
  
      .reveal section img {
          border: none;
          box-shadow: none;
          vertical-align: middle;
      }
  
      .reveal h1,
      .reveal h2,
      .reveal h3,
      .reveal h4,
      .reveal h5,
      .reveal h6 {
        font-family: "News Cycle", Lato, sans-serif;
        text-shadow: none; }
  
      div.bigpurple {
        font-size: 3em;
        line-height: 1em;
        color: purple;
      }
  
      div.caption {
        font-size: 24px;
        line-height: 1em;
      }
  
      div.citation {
          position: absolute;
          bottom:0px;
          right:0px;
          z-index:100;
          font-size: 0.45em !important;
      }
  
  </style>
</head>
<body>
  <div class="reveal">
    <div class="slides">

<section id="title-slide">
  <h1 class="title">Sequence alignment from the <br> phylogenetic perspective</h1>
  <p class="author">Erick Matsen</p>
</section>

<section id="in-phylogenetics-homology-means.." class="slide level1">
<h1>In phylogenetics, <em>homology</em> means..</h1>
<img src="figures/alignment/bigProtAln-column_nofonts.svg" class="stretch">
<div class="fragment">
Quiz: Why?
</div>
</section>
<section id="structural-alignment-is-another-perspective" class="slide level1">
<h1>Structural alignment is another perspective</h1>
<img src="figures/alignment/structural-alignment.png" class="stretch">
</section>
<section id="sequence-alignment-for-phylogenetics-is-hard" class="slide level1">
<h1>Sequence alignment for phylogenetics is hard!</h1>
<img src="figures/alignment/alignment-and-phylogeny-protpal-truth-only.png" class="stretch">
</section>
<section id="sometimes-its-more-or-less-impossible" class="slide level1">
<h1>Sometimes it’s more or less impossible</h1>
<img src="figures/alignment/alignment-and-phylogeny-protpal-hard-case.png" class="stretch">
</section>
<section id="phylogenetic-approach-is-best" class="slide level1">
<h1>Phylogenetic approach is best</h1>
<img src="figures/alignment/alignment-and-phylogeny-protpal-tractable-case.png" class="stretch">
<div class="citation">
<a href="http://dx.doi.org/10.1371/journal.pone.0034572">Westesson et al. 2012</a>
</div>
</section>
<section id="clustal-w-vs-prank" class="slide level1">
<h1>Clustal W vs PRANK</h1>
<img src="figures/alignment/alignment-and-phylogeny-prank.png" class="stretch">
<div class="citation">
<p><a href="http://dx.doi.org/10.1126/science.1158395">Löytynoja and Goldman 2008</a></p>
</div>
</section>
<section id="prank-is-a-smart-hack" class="slide level1">
<h1>PRANK is a smart “hack”</h1>
<img src="figures/alignment/prank.png" class="stretch">
<div class="citation">
<p><a href="http://dx.doi.org/10.1126/science.1158395">Löytynoja and Goldman 2008</a></p>
</div>
</section>
<section id="bali-phy-is-fully-model-based" class="slide level1">
<h1><a href="http://www.bali-phy.org/">BAli-Phy</a> is fully model-based</h1>
<img src="figures/alignment/bali-phy-pairwise-1_nofonts.svg" class="stretch">
<div class="citation">
<p><a href="http://dx.doi.org/10.1080/10635150590947041">Redelings &amp; Suchard 2005</a></p>
</div>
</section>
<section id="bali-phy-is-fully-model-based-1" class="slide level1">
<h1><a href="http://www.bali-phy.org/">BAli-Phy</a> is fully model-based</h1>
<img src="figures/alignment/bali-phy-pairwise-2_nofonts.svg" class="stretch">
<div class="citation">
<p><a href="http://dx.doi.org/10.1080/10635150590947041">Redelings &amp; Suchard 2005</a></p>
</div>
</section>
<section id="bali-phy-is-fully-model-based-2" class="slide level1">
<h1><a href="http://www.bali-phy.org/">BAli-Phy</a> is fully model-based</h1>
<img src="figures/alignment/bali-phy-pairwise-3_nofonts.svg" class="stretch">
<div class="citation">
<p><a href="http://dx.doi.org/10.1080/10635150590947041">Redelings &amp; Suchard 2005</a></p>
</div>
</section>
<section id="bali-phy-is-fully-model-based-3" class="slide level1">
<h1><a href="http://www.bali-phy.org/">BAli-Phy</a> is fully model-based</h1>
<img src="figures/alignment/bali-phy-pairwise_nofonts.svg" class="stretch">
<div class="citation">
<p><a href="http://dx.doi.org/10.1080/10635150590947041">Redelings &amp; Suchard 2005</a></p>
</div>
</section>
<section id="bali-phy-is-fully-model-based-4" class="slide level1">
<h1><a href="http://www.bali-phy.org/">BAli-Phy</a> is fully model-based</h1>
<img src="figures/alignment/bali-phy-pairwise_nofonts.svg" class="stretch">
<div class="citation">
<p><a href="http://dx.doi.org/10.1080/10635150590947041">Redelings &amp; Suchard 2005</a></p>
</div>
<p>Note how BAli-Phy can trace the history of each site through <br> the tree, and the columns are simply a reflection of that.</p>
</section>
<section id="bali-phy-estimates-alignment-uncertainty" class="slide level1">
<h1><a href="http://www.bali-phy.org/">BAli-Phy</a> estimates alignment uncertainty</h1>
<p>BAli-Phy is fully Bayesian, so it has a rigorous notion of alignment uncertainty, which can be elegantly summarized by that software:</p>
<img width="970" src="figures/alignment/bali-phy.png">
</section>
<section id="bali-phy-summary" class="slide level1">
<h1><a href="http://www.bali-phy.org/">BAli-Phy</a> summary</h1>
<ul>
<li>Co-estimate tree &amp; alignment using Bayesian methods</li>
<li>Can also use codon models for selection analysis; integrating out alignments <a href="http://dx.doi.org/10.1093/molbev/msu174">eliminates artifacts</a></li>
<li>Very computationally expensive: 60 sequences may take days</li>
</ul>
</section>
<section id="whats-funny-about-this-protein-coding-alignment" class="slide level1">
<h1>What’s funny about this protein-coding alignment?</h1>
<img src="figures/alignment/codon-1_nofonts.svg" class="stretch">
</section>
<section id="whats-funny-about-this-protein-coding-alignment-1" class="slide level1">
<h1>What’s funny about this protein-coding alignment?</h1>
<img src="figures/alignment/codon_nofonts.svg" class="stretch">
</section>
<section id="multiple-sequence-alignment-summary" class="slide level1">
<h1>Multiple sequence alignment summary</h1>
<p><br></p>
<ul>
<li class="fragment">
If you really care about alignments (and trees) use <a href="http://www.bali-phy.org/">BAli-Phy</a>
</li>
<li class="fragment">
<a href="https://mafft.cbrc.jp/alignment/software/">MAFFT</a> ranked the best in a recent <a href="http://dx.doi.org/10.1093/molbev/msy055">benchmark</a> by the authors of PRANK concerning ancestral sequence reconstruction
</li>
<li class="fragment">
If you want good alignment using <a href="http://dx.doi.org/10.1126/science.1158395">PRANK</a> <br> with a lovely web-based GUI, use <a href="http://wasabiapp.org/">Wasabi</a>
</li>
<li class="fragment">
If you want a quick and dirty alignment, use <a href="http://drive5.com/muscle/">Muscle</a> <br> (esp. if your sequences aren’t too divergent)
</li>
<li class="fragment">
Do not use any variant of Clustal, even if your friends say to.
</li>
<li class="fragment">
Use a codon alignment if you have protein-coding genes.
</li>
</ul>
</section>
    </div>
  </div>

  <script src="reveal.js/lib/js/head.min.js"></script>
  <script src="reveal.js/js/reveal.js"></script>

  <script>

      // Full list of configuration options available at:
      // https://github.com/hakimel/reveal.js#configuration
      Reveal.initialize({
        // Display controls in the bottom right corner
        controls: false,
        // Push each slide change to the browser history
        history: true,
        // Transition style
        transition: 'none', // none/fade/slide/convex/concave/zoom
        math: {
          mathjax: 'MathJax/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // Optional reveal.js plugins
        dependencies: [
          { src: 'reveal.js/lib/js/classList.js', condition: function() { return !document.body.classList; } },
          { src: 'reveal.js/plugin/zoom-js/zoom.js', async: true },
          { src: 'reveal.js/plugin/math/math.js', async: true },
          { src: 'reveal.js/plugin/notes/notes.js', async: true }
        ]
      });
    </script>
    </body>
</html>
